File: 02-week-2--recursion/05-day-5--recursive-selection-sort/javascript/selection_sort_recursive.js
function selectionSortRecursive(arr) {
  if (arr.length === 0) {
    return []
  }
  const min = Math.min(...arr)
  const index = arr.indexOf(min)
  arr.splice(index, 1)
  const order = selectionSortRecursive(arr)
  order.unshift(min)
  return order
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [-1, 2, 3, 5]");
  console.log("=>", selectionSortRecursive([3, -1, 5, 2]));

  console.log("");
}

module.exports = selectionSortRecursive;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 02-week-2--recursion/05-day-5--recursive-selection-sort/ruby/selection_sort_recursive.rb
def selection_sort_recursive(arr)
  return [] if arr.empty?
  min = arr.min
  index = arr.index(min)
  arr.delete_at(index)
  order = selection_sort_recursive(arr)
  order.unshift(min)
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort_recursive([3, -1, 5, 2])

  puts

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
